Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs

```diff
@@ -9,38 +10,155 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Primitives;
+using System.Net;
 
 namespace Microsoft.DotNet.DarcLib.Tests
 {
-    class GitHubClientTests
+    #region Fakes
+    public class SimpleCacheEntry : ICacheEntry
     {
-        #region GitHubClient with set-able IGitHubClient 
-        /// <summary>
-        /// Lacking any DI, this class lets us put a Mock IGitHubClient into something that is effectively the same,
-        /// other than providing the ability to stick any IGitHubClient in as desired.
-        /// </summary>
-        class TestGitHubClient : GitHubClient
+        private object _key;
+        private object _value;
+        private long? _size;
+
+        public SimpleCacheEntry(object key)
+        {
+            _key = key;
+        }
+
+        public object Key => _key;
+
+        public object Value { get => _value; set => _value = value; }
+        public DateTimeOffset? AbsoluteExpiration { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public TimeSpan? AbsoluteExpirationRelativeToNow { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public TimeSpan? SlidingExpiration { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public IList<IChangeToken> ExpirationTokens => throw new NotImplementedException();
+
+        public IList<PostEvictionCallbackRegistration> PostEvictionCallbacks => throw new NotImplementedException();
+
+        public CacheItemPriority Priority { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public long? Size { get => _size; set => _size = value; }
+
+        public void Dispose() { }
+    }
+
+    public class SimpleCache : IMemoryCache
+    {
+        public int CacheHits { get; set; }
+        public int CacheMisses { get; set; }
+        public ConcurrentDictionary<object, ICacheEntry> cache = new ConcurrentDictionary<object, ICacheEntry>();
+
+        public ICacheEntry CreateEntry(object key)
         {
-            private IGitHubClient _client;
-            public void SetGitHubClientObject(IGitHubClient value)
+            var newEntry = new SimpleCacheEntry(key);
+            return cache.AddOrUpdate(key, new SimpleCacheEntry(key), (existingKey, existingValue) => newEntry);
+        }
+
+        [TearDown]
```

Comment:
Why does this have `[TearDown]` on it? Its just the Dispose method that implements the IDisposable interface, and it doesn't do anything because our implementation doesn't have anything to dispose. Isn't `[TearDown]` only for methods in the test classes?

Reply:
This was Chad's code from the other file that copied into this one. But yes, I believe that the TearDown attribute is only for tests.